classes/local/execution/iterators/php_iterator.php

```
@@ -79,6 +79,15 @@ public function next() {
         if ($this->finished) {
             return false;
         }
+
+        // Only performs this check on the first iteration, and aborts if the
+        // iterator's position is valid (e.g. has no data).
+        if ($this->iterationcount === 0 && !$this->input->valid()) {
+            $this->step->log('Aborting at iteration ' . $this->iterationcount . '. No data?');
+            $this->abort();
```

I'm not certain what your intention is here, but as an FYI, calling an iterator's abort( ) function (I thinking it's a bad choice of name), will not end the dataflow. If you intend to terminate the dataflow, you need to throw an exception. The engine will pick it up and abort all steps and iterators.

---

I don't necessarily think an empty flow should terminate the whole dataflow, which is why I've opted for a similar check to the one further down. The issue comes when the value provided by the first iteration is invalid, because the records returned initially is empty.

e.g. instead of something like `{id: 1, name: "Kevin"}` it's returning `{"scalar": []}` or similar

---

Fair enough, as long as this is intentional, and not an accident.

---

To be clear I expect it to be quite normal for an iterator in real life to have no records and thats fine. eg you might make a curl to grab a delta, turns that into iterator, has nothing in it, and then a connector at the end which sends an email which says 'there was 0 records processed'. At no point will there be an error or abort